MaxText/layers/attentions.py

```diff
@@ -528,7 +528,25 @@ def kv_cache_prefill(
     cached_prefill_key_var, cached_prefill_value_var, cached_prefill_segment_id = self._get_prefill_cache(
         batch, heads, kv_head_size, self.quantize_kvcache
     )
-    self._get_ar_cache(batch, heads, kv_head_size, self.quantize_kvcache)  # initialize it now
+    cached_ar_key_var, cached_ar_value_var, _, _ = self._get_ar_cache(batch, heads, kv_head_size, self.quantize_kvcache)  # initialize it now
+    cached_ar_key_var[0].value = nn.with_logical_constraint(
```

Comment:
please put this into the `self._get_ar_cache` but looks ok otherwise as a short term fix.

Reply:
done